src/urllib3/contrib/ssl.py

```diff
@@ -0,0 +1,173 @@
+import ssl
+import socket
+import io
+
+
+class SSLTransportError(Exception):
+    "Will wrap any SSL errors received during socket IO."
+    pass
+
+
+class SSLTransport:
```

Comment:
How will this class work with pyOpenSSL and SecureTransport SSLContext objects?

Reply:
This is a good point and something I wasn't considering. I'll double-check if it's compatible.

Reply:
Update here:

This doesn't support pyOpenSSL or SecureTransport SSLContext. SecureTransport doesn't provide the ``wrap_bio`` methods we need to create the second TLS layer. pyOpenSSL does but I would need to implement it. I hope that, for now, adding support for just ``ssl`` is sufficient.

Reply:
This is fine, it's good that we got Requests using Python's TLS implementation by default now.